numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -155,11 +155,15 @@ npy_PyFile_Dup2(PyObject *file, char *mode, npy_off_t *orig_pos)
     /* Flush first to ensure things end up in the file in the correct order */
     ret = PyObject_CallMethod(file, "flush", "");
     if (ret == NULL) {
+        PyErr_SetString(PyExc_IOError,
```

Comment:
The error should have been set by the call to `file.flush`, I don't think this is needed.